codersdk/deployment.go

```diff
@@ -942,6 +953,33 @@ when required by your organization's security policy.`,
 			Group:       &deploymentGroupIntrospectionPrometheus,
 			YAML:        "collect_agent_stats",
 		},
+		{
+			Name:        "Prometheus Aggregate Agent Stats By",
+			Description: fmt.Sprintf("When collecting agent stats, aggregate metrics by a given set of comma-separated labels to reduce cardinality. Accepted values are %q.", AcceptedMetricAggregationLabels),
+			Flag:        "prometheus-aggregate-agent-stats-by",
+			Env:         "CODER_PROMETHEUS_AGGREGATE_AGENT_STATS_BY",
+			Value: clibase.Validate(&c.Prometheus.AggregateAgentStatsBy, func(value *clibase.StringArray) error {
+				if value == nil {
+					return nil
+				}
+
+				acceptable := make(map[string]any, len(AcceptedMetricAggregationLabels))
```

Comment:
I think using `map[string]bool` would be preferable/clearer here, also simplifies the map lookup later. Typically I'd use either `bool` or if we're looking for space savings, I'd use the zero struct (`map[string]struct{}`).